src/helpers/figures/charts/scatter_chart.ts

```diff
@@ -0,0 +1,202 @@
+        : undefined,
+      title: this.title,
+      labelsAsText: this.labelsAsText,
+      aggregated: this.aggregated,
+    };
+  }
+
+  getContextCreation(): ChartCreationContext {
+    return {
+      background: this.background,
+      title: this.title,
+      range: this.dataSets.map((ds: DataSet) =>
+        this.getters.getRangeString(ds.dataRange, this.sheetId)
+      ),
+      auxiliaryRange: this.labelRange
+        ? this.getters.getRangeString(this.labelRange, this.sheetId)
+        : undefined,
+    };
+  }
+
+  updateRanges(applyChange: ApplyRangeChange): ScatterChart {
+    const { dataSets, labelRange, isStale } = updateChartRangesWithDataSets(
+      this.getters,
+      applyChange,
+      this.dataSets,
+      this.labelRange
+    );
+    if (!isStale) {
+      return this;
+    }
+    const definition = this.getDefinitionWithSpecificDataSets(dataSets, labelRange);
+    return new ScatterChart(definition, this.sheetId, this.getters);
+  }
+
+  getDefinitionForExcel(): ExcelChartDefinition | undefined {
+    return undefined; // TODO
+  }
+
+  copyForSheetId(sheetId: UID): ScatterChart {
+    const dataSets = copyDataSetsWithNewSheetId(this.sheetId, sheetId, this.dataSets);
+    const labelRange = copyLabelRangeWithNewSheetId(this.sheetId, sheetId, this.labelRange);
+    const definition = this.getDefinitionWithSpecificDataSets(dataSets, labelRange, sheetId);
+    return new ScatterChart(definition, sheetId, this.getters);
+  }
+
+  copyInSheetId(sheetId: UID): ScatterChart {
+    const definition = this.getDefinitionWithSpecificDataSets(
+      this.dataSets,
+      this.labelRange,
+      sheetId
+    );
+    return new ScatterChart(definition, sheetId, this.getters);
+  }
+}
+
+export function createScatterChartRuntime(
+  chart: ScatterChart,
+  getters: Getters
+): ScatterChartRuntime {
+  const { chartJsConfig, background, dataSetsValues, labelValues, dataSetFormat, labelFormat } =
+    createLineOrScatterChartRuntime(chart, getters);
+  chartJsConfig.type = "line";
+
+  const configOptions = chartJsConfig.options!;
+  configOptions.elements = {
+    point: {
+      radius: 3,
+      hoverRadius: 3, // chartJS seems bugged, the point starts with the "hoverRadius" value
+      hitRadius: 8,
+    },
+  };
+
+  const locale = getters.getLocale();
+
+  configOptions.plugins!.tooltip!.callbacks!.title = () => "";
+  configOptions.plugins!.tooltip!.callbacks!.label = (tooltipItem) => {
+    const dataSetPoint = dataSetsValues[tooltipItem.datasetIndex!].data![tooltipItem.dataIndex!];
+    let label: string | number = labelValues.values[tooltipItem.dataIndex!];
+    if (isNumber(label, locale)) {
+      label = toNumber(label, locale);
+    }
+    const formattedX = formatValue(label, { locale, format: labelFormat });
+    const formattedY = formatValue(dataSetPoint, { locale, format: dataSetFormat });
+    const dataSetTitle = tooltipItem.dataset.label;
+    return `${dataSetTitle}: (${formattedX}, ${formattedY})`;
```

Comment:
I'm not sure what we want in this situation but there can be ´undefined´ in here.
In the situation described in the screenshot below, hover the last two data points: `"Series 1: (undefined, 2)"`
![image](https://github.com/odoo/o-spreadsheet/assets/10908305/2868af59-ee1c-4caf-a445-4144eba51be6)